### PR TITLE
AB#51487 choicesByUrl not working with new metadata

### PR DESCRIPTION
**Pull request by ketanreliefapps**
# Description
Cannot set filters on a choicesByUrl field.

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (refactor or addition to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Create field drop-down with choicesByUrl and using that fields as filter, so now you can see filter value

## Sreenshots
image

# Checklist:
- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/schema/query/resource.query.ts [hunk 1]
```diff
@@ -22,6 +23,18 @@ export default {
     const ability: AppAbility = user.ability;
     const resource = await Resource.findOne({ _id: args.id });
 
+    let index = 0;
+    for await (const field of resource.fields) {
```
Comment (AntoineRelief) [inline src/schema/query/resource.query.ts:27]: @ketanreliefapps we should not put that there, but in the resolver of the resource.type + form.type

Comment (AntoineRelief) [reply src/schema/query/resource.query.ts:27]: I think the logic is good, but it cannot work for example if you query multiple resources at same time

Comment (ketanreliefapps) [reply src/schema/query/resource.query.ts]: i have added in the meta resolver

File: src/utils/form/metadata.helper.ts [hunk 2]
```diff
@@ -229,21 +230,37 @@ export const getMetaData = async (
...
-          canUpdate: ability.can('update', parent, `data.${field.name}`),
-          ...(field.choices && {
-            options: field.choices.map((x) => {
-              return {
-                text: x.text ? x.text : x,
-                value: x.value ? x.value : x,
-              };
+        if (field.choicesByUrl) {
+          const fieldData = field;
+          fieldData.choicesByUrl.value = '';
```
Comment (AntoineRelief) [inline src/utils/form/metadata.helper.ts:235]: @ketanreliefapps not sure why you remove the values there ?

**Pull request merged**